gr/lib/buffer_management.cc

```diff
@@ -78,18 +81,20 @@ void buffer_manager::initialize_buffers(flat_graph_sptr fg,
                     d_debug_logger->debug(
                         "Creating Buffer Reader for Edge: {}, Independently",
                         ed[0]->identifier());
-                    p->set_buffer_reader(ed[0]->buf_properties()->reader_factory()(
-                        ed[0]->dst().port()->itemsize(), ed[0]->buf_properties()));
-                    p->buffer_reader()->set_parent_intf(sched_intf);
+                    auto br = ed[0]->buf_properties()->reader_factory()(
+                        ed[0]->dst().port()->itemsize(), ed[0]->buf_properties());
+                    p->set_buffer_reader(br);
+                    p->get_buffer_reader()->set_parent_intf(sched_intf);
                 }
                 else {
 
                     d_debug_logger->debug(
                         "Adding Buffer Reader for Edge: {}, to buffer on Block {}",
                         ed[0]->identifier(),
                         ed[0]->src().identifier());
-                    p->set_buffer_reader(ed[0]->src().port()->buffer()->add_reader(
-                        ed[0]->buf_properties(), ed[0]->dst().port()->itemsize()));
+                    auto br = ed[0]->src().port()->get_buffer()->add_reader(
+                        ed[0]->buf_properties(), ed[0]->dst().port()->itemsize());
+                    p->set_buffer_reader(br);
```

Comment:
Even though I already added the comments about `&&` this confused me. :-)

`br` is a smart pointer, but reading the code here it looks like it does not pass ownership (because there's no `std::move`). By having `add_reader` take `&&` and forcing a `std::move` here, it makes the ownership change clear.